### PR TITLE
Project 4.4

### PR DESCRIPTION
**Pull request by Daniel-Martin576**
Project 4 for learning to implement PID controller
I pretty much copied the exercise from `ros-exercises` with some edits. Let me know if you want me to change the markdown more or if the exercise is too involved. I used `buzzsim` for this exercise but I could remove it.

### CODE REVIEW

File: assignments/week_4/week_4_exercises/project_4_4/README.md [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+Let's try a moving target this time. Do this by using Ctrl-/ and commenting the `<param name="world_name" value="stationary" />` line
+and uncommenting the line below in the [project_4_4.launch](../launch/project_4_4.launch) file.
+
+It should look like this:
+```launch
+<launch>
+    <node pkg="buzzsim" type="buzzsim" name="buzzsim">
+        <param name="config_path" value="$(find buzzsim)/config/world.yml" />
+        <param name="world_name" value="stationary" />
+       <!-- <param name="world_name" value="moving" /> -->
```
Comment (matthewhannay567) [inline assignments/week_4/week_4_exercises/project_4_4/README.md]: Wrong parameter is commented out

**Review by matthewhannay567 — COMMENTED** [review]
I'm concerned about the usage of launch files and parameters, as we intend to cover that next week.

File: assignments/week_4/week_4_exercises/project_4_4/README.md (hunk 1 above)
Comment (Daniel-Martin576) [reply assignments/week_4/week_4_exercises/project_4_4/README.md]: How so? I want them to do `stationary` first

**Comment by Daniel-Martin576** [comment]
Valid concern. However, exercise 4.1 and 4.3 use launch files and the alternative is that they hard code the Kp, Ki, Kd values in the code which I don't think we want.

File: assignments/week_4/week_4_exercises/project_4_4/README.md (hunk 1 above)
Comment (matthewhannay567) [reply assignments/week_4/week_4_exercises/project_4_4/README.md]: This is supposed to be the launch code example for when you want it to be `moving`

Comment (Daniel-Martin576) [reply assignments/week_4/week_4_exercises/project_4_4/README.md]: I want to run PID controller when the top turtle is stationary and then run it when the top turtle is moving.

Comment (matthewhannay567) [reply assignments/week_4/week_4_exercises/project_4_4/README.md]: Yes... and this is the example launch code after you say they should be "commenting the `<param name="world_name" value="stationary" />` line and uncommenting the line below", but it's still has the "stationary" parameter uncommented and the "moving" one commented.

Comment (Daniel-Martin576) [reply assignments/week_4/week_4_exercises/project_4_4/README.md]: I understand now. Fixed

**Comment by Daniel-Martin576** [comment]
Should I take in Kp, Ki, Kd, and world_type from a file/cin or keep the launch file format?

**Comment by matthewhannay567** [comment]
Maybe have them hard-coded for now, and this can sort of teach the new members that hard-coded values are a pain to deal with, and next week they learn that launch parameters save a lot of time.

**Comment by Daniel-Martin576** [comment]
Updated. I removed all the launch file parameter stuff

**Review by matthewhannay567 — APPROVED** [review]
Looks good to me

**Pull request merged**